packages/visualizations/src/components/Table/types.ts

```diff
@@ -34,42 +34,65 @@ type BaseColumn = {
     onClick?: () => void;
 };
 
-// Eliminate rawValue from props to match column options
-type ColumnOptions<T> = Omit<T, 'rawValue'>;
+export type ValueOrAccessor<T, R> = T | ((r: R) => T);
+
+export type FormatPropsTypeMap = {
+    [DATA_FORMAT.boolean]: BooleanFormatProps;
+    [DATA_FORMAT.date]: DateFormatProps;
+    [DATA_FORMAT.geo]: GeoFormatProps;
+    [DATA_FORMAT.shortText]: TextFormatProps;
+    [DATA_FORMAT.longText]: TextFormatProps;
+    [DATA_FORMAT.number]: NumberFormatProps;
+    [DATA_FORMAT.url]: URLFormatProps;
+};
 
-type FormatPropsTypeMap = {
-    [DATA_FORMAT.boolean]: ColumnOptions<BooleanFormatProps>;
-    [DATA_FORMAT.date]: ColumnOptions<DateFormatProps>;
-    [DATA_FORMAT.geo]: ColumnOptions<GeoFormatProps>;
-    [DATA_FORMAT.shortText]: ColumnOptions<ShortTextFormatProps>;
-    [DATA_FORMAT.longText]: ColumnOptions<LongTextFormatProps>;
-    [DATA_FORMAT.number]: ColumnOptions<NumberFormatProps>;
-    [DATA_FORMAT.url]: ColumnOptions<URLFormatProps>;
+export type ReturnTypeMap = {
+    [DATA_FORMAT.boolean]: boolean;
+    [DATA_FORMAT.date]: string;
+    [DATA_FORMAT.geo]: WebGlMapData;
+    [DATA_FORMAT.shortText]: string;
+    [DATA_FORMAT.longText]: string;
+    [DATA_FORMAT.number]: number;
+    [DATA_FORMAT.url]: string;
 };
 
-export type ColumnOfType<F extends DataFormat> = BaseColumn & {
+/** Columns have to be typed with the record type if using an accessor.
+ * They can also be used without a record, nor accessor
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type ColumnOfType<F extends DataFormat, R = any> = BaseColumn & {
     dataFormat: F;
-    options?: FormatPropsTypeMap[F];
+    accessor?: (r: R) => ReturnTypeMap[F];
+    options?: ValueOrAccessor<Omit<FormatPropsTypeMap[F], 'value'>, R>;
```

Comment:
```suggestion
    accessor?: (r: R) => FormatPropsTypeMap[F]['value'];
    options?: ValueOrAccessor<Omit<FormatPropsTypeMap[F], 'value'>, R>;
```

With that `ReturnTypeMap` can be removed